### ISSUE TITLE
[Bug]: Multiple Project Folders Lack requirements.txt — Reproducibility Is Broken for Learners

### ISSUE THREAD

**Issue by divyanshim27** [Niketkumardheeryan/ML-CaPsule#2299]
## Summary

This is a concrete, auditable bug affecting ML-CaPsule's core value as a learning resource. A significant number of project folders — including some of the most popular ones — do not contain a `requirements.txt` file. When a learner clones the repo and tries to run a notebook, they encounter `ModuleNotFoundError` with no guidance on what to install.

## Affected Projects (Non-exhaustive Sample)

Based on a scan of the repository tree, the following folders appear to be missing `requirements.txt`:

- `Bidirectional LSTM`
- `GRAD-CAM-Visualizer`
- `GUI-JARVIS`
- `Gesture Control Mouse`
- `Digital Image Tampering Detection`
- `Dynamic Hedging Strategies using Reinforcement Learning`
- `Generating 3D Design Voxels using GANs`

(A full audit will be performed as part of the fix.)

## Steps to Reproduce

1. Clone the repo: `git clone https://github.com/Niketkumardheeryan/ML-CaPsule`
2. Navigate to `cd "Bidirectional LSTM"`
3. Attempt to run: `pip install -r requirements.txt` — **File not found**
4. Open the notebook and run — receive `ModuleNotFoundError` for `tensorflow`, `keras`, etc.

## Impact

- Directly breaks the learning experience for the target audience (beginners without ML environment setup experience).
- Increases GitHub Issues noise as learners open "how do I install this?" questions.
- Reduces the repo's credibility as a production-quality educational resource.

## Proposed Solution

1. Audit every project folder using a Python script to detect missing `requirements.txt`:

```python
import os

missing = []
for folder in os.listdir('.'):
    if os.path.isdir(folder) and not folder.startswith('.'):
        if 'requirements.txt' not in os.listdir(folder):
            missing.append(folder)

print(f"Missing requirements.txt in {len(missing)} folders:")
for m in missing: print(f"  - {m}")
```

2. For each folder, add a `requirements.txt` with pinned versions by inspecting the notebook imports.

3. Add a CI check via GitHub Actions that fails if any new project folder PR does not include `requirements.txt`.

I will run the full audit and submit a PR with all missing files added. Could you assign this to me?

**Labels:** `bug`, `good first issue`, `documentation`, `reproducibility`, `GSSoC 2026`
